src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -974,6 +1022,43 @@ public ListBoxModel doFillLocationItems(@RelativePath("..") @QueryParameter Stri
             return model;
         }
 
+        public String doFillAvailabilityTypeItems() {
+            return null;
+        }
+
+        public ListBoxModel doFillAvailabilitySetItems(
+                @RelativePath("../..") @QueryParameter String azureCredentialsId,
+                @RelativePath("../..") @QueryParameter String resourceGroupReferenceType,
+                @RelativePath("../..") @QueryParameter String newResourceGroupName,
+                @RelativePath("../..") @QueryParameter String existingResourceGroupName,
+                @RelativePath("..") @QueryParameter String location) {
+            ListBoxModel model = new ListBoxModel();
+            model.add("--- Select Availability Set in current resource group and location ---", "");
```

Comment:
if  possible to list availability in another resource group within same subs?

Reply:
In current design, all agents resources are in the same resource group. Does it necessary to provide an option here for users to create a new availability?